### ISSUE TITLE
Add `diesel drop` to the CLI

### ISSUE THREAD

**Issue by mcasper** [sgrif/diesel#146]
We should have some way of dropping the user's database in the CLI, for whenever the user wants to reset their data or their schema. This command could be `diesel drop`, `diesel drop_db`, `diesel drop_database`... This would allow the workflow of `diesel drop; diesel setup` to reset your database. 

My thought of what it should do:
* Drop the database specified in the user's database-url string.

If we like this, I'll go ahead and implement it this afternoon.

**Comment by mcasper** [comment]
Or maybe database should be a subcommand, and then have its own derivatives. So it might become
```
diesel database drop
diesel database setup
diesel database foo
```

**Comment by sgrif** [comment]
Is there any use case for just `drop` on its own? Should we just have this be `diesel database reset`?

**Comment by mcasper** [comment]
You know, I can't immediately come up with a use case for `drop` on its own. Maybe just `reset` for now, and see if legitimate use cases for `drop` come in

**Comment by sgrif** [comment]
Just to clarify, my reasoning here is that we definitely want `reset` as a command either way, and I'd rather not expose `drop` on its own without a specific reason, as it's just one more thing to maintain and/or change if we ever decide to do something like https://github.com/rails/rails/pull/22967

**Comment by mcasper** [comment]
Resolved by #161
